apm-agent-core/src/main/java/co/elastic/apm/agent/report/serialize/Base64SerializationUtils.java

```diff
@@ -0,0 +1,43 @@
+package co.elastic.apm.agent.report.serialize;
+
+import com.dslplatform.json.JsonWriter;
+
+public class Base64SerializationUtils {
+
+    private static final byte[] BASE64_URL_CHARS = new byte[]{
+        'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H',
+        'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P',
+        'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X',
+        'Y', 'Z', 'a', 'b', 'c', 'd', 'e', 'f',
+        'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n',
+        'o', 'p', 'q', 'r', 's', 't', 'u', 'v',
+        'w', 'x', 'y', 'z', '0', '1', '2', '3',
+        '4', '5', '6', '7', '8', '9', '-', '_',
+    };
+
+
+    public static void writeBytesAsBase64UrlSafe(byte[] data, JsonWriter jw) {
```

Comment:
The java `Base64` class used in the otel-distro is only available in Java 8+, that's why we need to add a custom implementation here. Also this implementation is allocation free, keeping it in line with the rest of the agent design.